### ISSUE TITLE
Poor connection quality

### ISSUE THREAD

**Issue by zzxx53** [jitsi/jitsi-meet#2191]
We have tried out the meet.jit.si demo page could get conference connected. However, the call quality deteriorates quickly, usually seconds into a call. The connection indicator goes from green to red, and video freezes frequently. Speed test indicates we have >50Mb/s download and >20Mb/s upload, so network shouldn't be the bottleneck. Also, the same issue doesn't happen on other solutions (FrozenMountain LiveSwitch, appear.in).  
We thought this might be due to limited capacity on the demo server, so we set up our own instance on AWS, only to find the results to be very similar, and the CPU and memory usage on the AWS server is pretty low (2~3% CPU, ~1GB RAM used). 
Is there anything else that we can look at? Any suggestions are welcome.

**Comment by saghul** [comment]
Can you check if any packet loss is detected? You should be able to see that in the panel that shows up when you hover on the connection indication for yourself, ie, the one at the top.

**Comment by zzxx53** [comment]
Packet loss goes up and down, but generally not more than 5%. This happens for both uploading and downloading packets.

**Comment by zzxx53** [comment]
Got a stable 5-party call with video height constraints 360 specified in config.js. Guessing this is a network issue. 
Are there any recommendations you can provide about making the connections more stable? Would changing EC2 instance type help with the issue?

**Comment by gpolitis** [comment]
Hi @zzxx53, can I test your deployment somewhere?

**Comment by zzxx53** [comment]
@gpolitis My current deployment is at https://jitsi.liveexpertteset.net . I took out the lines for restricting video height so that it is running with default config.

**Comment by gpolitis** [comment]
Do you have a typo there? The domain name doesn't resolve.

**Comment by zzxx53** [comment]
@gpolitis corrected the address; thanks for testing this

**Comment by gpolitis** [comment]
I started a 1-1 call and I got routed through the bridge and on TCP, which is the worst possible combination you can have. Have you opened/redirected the UDP ports on your instance? Also, I would enable p2p mode. Try these two suggestions and please let us know if it works any better.

**Comment by zzxx53** [comment]
Opened UDP ports 10000-20000; didn't see much improvement in call quality. 
What would be a definite way to know that the call is running through UDP? I checked netstat on the server and saw no established UDP connections. 
![image](https://user-images.githubusercontent.com/1800269/33098895-85e14ba2-ced4-11e7-801d-ec2fba2dd4d1.png)

**Comment by damencho** [comment]
You need to forward those ports in aws console, it was in the security group somewhere. The same way you did for port 443.

**Comment by gpolitis** [comment]
You mentioned EC2, so you need to configure and assign the appropriate security groups to your instance.

**Comment by zzxx53** [comment]
This is my current security group: 
![image](https://user-images.githubusercontent.com/1800269/33099228-cfa76fae-ced5-11e7-9c1a-7be7c94dc3d5.png)
So ports 10000-20000 should be forwarded. 
And the call quality appears to be equally bad when I'm at the network where I have access to all UDP ports

**Comment by gpolitis** [comment]
UDP appears to be working now. I've run a quick 3-party test and was able to get HD on all participants. What's in your JVB sip-communicator.properties file?

**Comment by zzxx53** [comment]
Just the line from the guide: `org.jitsi.impl.neomedia.transform.srtp.SRTPCryptoContext.checkReplay=false`

**Comment by TheReal1604** [comment]
@zzxx53 @gpolitis got the same results also with a attendee which got a bit of packet loss. My jitsi instance is hosted on a physical server.

**Comment by openaspace** [comment]
Hello.
Same problem here...
how is possible? 

I'm running the debian server under virtualbox with 4 core 3.2ghz and 16gb of ram on a sas disk at 15k rpm on my home fiber connection at 1gbps down 100mbps up.

Thank-you.

**Comment by GreenFingers** [comment]
Any update on how to fine-tune or debug a poorly-performing Jitsi Video Bridge?

**Comment by TheReal1604** [comment]
@saghul @damencho maybe? :) I wrote with damencho some days before - there is an upload requirement of around 2mbit~ (and udp should not be firewalled or something) which the attendees should have to get high quality.

**Comment by GreenFingers** [comment]
@damencho Jitsi videobridge is set up in a virtual machine on a cloud provider and I am testing with a single call going over the server using UDP (let's assume p2p is not an option). Resolution on jitsi meet is set quite low (540 with 16:9 aspect ratio). However, the image freezes on both participant's screens after a while and lag builds up afterwards. The server connection is quite good (500 Mbps) and the client connections as well. Processor and memory usage on the VM never go over 20%. When using the same clients with meet.jit.si the connection is fluid. 

This leads me to think:

- [ ] Network connections are good
- [ ] Clients are capable of handling WebRTC without lag
- [ ] VM processor/CPU is not a bottleneck

I begin to suspect this could be due to the virtual interfaces of my VM. However tcpdump at the server shows that only a couple of UDP packets are being lost during the conversation. 

Do you have any ideas on how to debug this, or if there are any parameters that should be fine-tuned in the operating system or jitsi-videobridge?

Thanks!

**Comment by damencho** [comment]
Are you installing jitsi-videobridge from debian package? There are some settings that are done when installation: https://github.com/jitsi/jitsi-videobridge/blob/master/config/20-jvb-udp-buffers.conf

**Comment by GreenFingers** [comment]
Hi @damencho, thanks for pointing that out. Any idea why other parameters, such as **net.core.wmem_max** are not increased as well? Are the values provided in the conf file fine-tuned for some specific scenario?

**Comment by damencho** [comment]
@GreenFingers I don't know details. These values were tuned at some reasonable values so we stop seeing errors on OS level on meet.jit.si.
@gpolitis @bgrozev @aaronkvanmeerten ^ do you have more info on the subject?

**Comment by TheReal1604** [comment]
@damencho just a feeling. It is possible that one attendee can lower the overall quality if he has a bad connection? At some tests on my deployment the connection was bad for all if one specific user with a bad connection joined.

**Comment by openaspace** [comment]
I don't really understand why my system stop video for low bandwidth ... if it run on 100mbps down and 50mbps up speed...

**Comment by havid0707** [comment]
I also encountered this problem.  ```It is possible that one attendee can lower the overall quality if he has a bad connection?```

Frequently prompting a user to having connectivty issues..

`const isConnectionInterrupted
                = APP.conference.getParticipantConnectionStatus(id)
                    === JitsiParticipantConnectionStatus.INTERRUPTED;
            let messageKey = null;

            if (isConnectionInterrupted) {
                messageKey = 'connection.USER_CONNECTION_INTERRUPTED';
            } else if (connectionStatus
                    === JitsiParticipantConnectionStatus.INACTIVE) {
                messageKey = 'connection.LOW_BANDWIDTH';
            }`

**Comment by havid0707** [comment]
@damencho 
![image](https://user-images.githubusercontent.com/9440277/44706058-fda36c80-aad2-11e8-940f-f8daa3e52cde.png)

I can not find this file ```20-jvb-udp-buffers.conf``` in my jvb installation, where should I add these parameters ?

/etc/jitsi/videobridge
/usr/share/jitsi-videobridge

**Comment by saghul** [comment]
IIRC that file is in `/etc/sysctl.d` or thereabouts.

**Comment by havid0707** [comment]
Yes, I found the file, then it seems that I have nothing to do for this case.
Are there other parameters for solving the video freeze problem ?
Frequently prompting users to have connection problems makes the entire user experience worse

**Comment by havid0707** [comment]
Yes

**Comment by damencho** [comment]
So this most probably means the connectivity of the clients is not good, so there is nothing you can do serverside.

**Comment by Christilut** [comment]
For anyone finding this, the solution for me was to disable `SIP ALG` in my router. Reconnected with Jitsi and connection went from Poor to Good.
